lib/src/elf.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as fs from 'fs';
+
+/**
+ * Read chunk of data from a file descriptor into a new Buffer.
+ */
+const readFileDescriptor = function (
+  fd: number,
+  position: number,
+  length: number
+): Buffer {
+  const buffer = Buffer.alloc(length);
+  let offset = 0;
+  while (offset < length) {
+    const bytesRead = fs.readSync(fd, buffer, {
+      offset: offset,
+      position: position + offset,
+    });
+    if (bytesRead === 0) {
+      throw new Error(`failed to read fd ${fd}`);
+    }
+
+    offset += bytesRead;
+  }
+  return buffer;
+};
+
+/**
+ * Parse an ELF file and return its interpreter.
+ */
+export const getELFInterpreter = function (path: string): string {
+  const fd = fs.openSync(path, 'r');
+  try {
+    const eIdent = new DataView(readFileDescriptor(fd, 0, 64).buffer);
```

Comment:
What does this `e` indicate here?

Reply:
In original kernel code this is named as `e_ident` (I think e stands for ELF):
 https://github.com/torvalds/linux/blob/21b136cc63d2a9ddd60d4699552b69c214b32964/include/uapi/linux/elf.h#L210C17-L210C24